lib/app.js

```diff
@@ -22,12 +22,6 @@ export default class App extends Component {
     }
   }
 
-  componentDidCatch (error, info) {
```

Comment:
Why we remove this?

Reply:
1. it's not SSR compatible and we didn't add it (community contribution for React v16 support)
2. It didn't do anything. Tried multiple times and the error component got rendered just fine.

Reply:
`if (this.state.hasError) return null` should probably remove this too 😄

Reply:
@timneutkens for SSR we don't need this. We can easily catch them with try catch. This is for development to track errors inside render functions.
(Or it's more for tracking an edge case)

Earlier, we had to hack React to get this. Please do not remove it.
Create an issue asking whether we really need this.
Sometimes, React Hot Loader might already use it. If so we can remove this.
Otherwise, it's better to keep it.

Reply:
@timneutkens This is the hack we had.
See: https://github.com/zeit/next.js/pull/2996/files#diff-1858c8296dd2b642c74271609cf701e7L39

`componentDidMount` just replaced it.